cmd/snapshot-decoder/main.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+// Package main implements a simple tool to decode a snapshot file.
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	storagepb "github.com/siderolabs/discovery-service/api/storage"
+)
+
+var snapshotPath = "/var/discovery-service/state.binpb"
+
+func init() {
+	flag.StringVar(&snapshotPath, "snapshot-path", snapshotPath, "path to the snapshot file")
+}
+
+func main() {
+	flag.Parse()
+
+	if err := run(); err != nil {
+		log.Fatalf("error: %v", err)
+	}
+}
+
+func run() error {
+	data, err := os.ReadFile(snapshotPath)
+	if err != nil {
+		return fmt.Errorf("failed to read snapshot: %w", err)
+	}
+
+	snapshot := &storagepb.StateSnapshot{}
+
+	if err = snapshot.UnmarshalVT(data); err != nil {
+		return fmt.Errorf("failed to unmarshal snapshot: %w", err)
+	}
+
+	encoder := json.NewEncoder(os.Stdout)
```

Comment:
we might consider https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson